favor-state option support for --defer (#2676)
fishtown-analytics/docs.getdbt.com: website/docs/reference/node-selection/defer.md

</Changelog>

> Deferral is a powerful, complex feature that enables compelling workflows. As the use cases for `--defer` evolve, we could make enhancements to the feature, but commit to providing backward compatibility for supported versions of dbt Core.  For details, see [dbt#5095](https://github.com/dbt-labs/dbt-core/discussions/5095).

Reviewer: ```suggestion
Deferral is a powerful, complex feature that enables compelling workflows. As the use cases for `--defer` evolve, dbt Labs could make enhancements to the feature, but commit to providing backward compatibility for supported versions of dbt Core.  For details, see [dbt#5095](https://github.com/dbt-labs/dbt-core/discussions/5095).
```

> **N.B.** Deferral is a powerful, complex feature that enables compelling workflows. As the use cases for `--defer` evolve, enhancements may be made to the feature, but we are committed to providing backward compatibility for supported versions of dbt Core.  For details, see [dbt#5095](https://github.com/dbt-labs/dbt-core/discussions/5095).

Reviewer 2: Just making this active.
```suggestion
**N.B.** Deferral is a powerful, complex feature that enables compelling workflows. As the use cases for `--defer` evolve, we could make enhancements to the feature, but commit to providing backward compatibility for supported versions of dbt Core.  For details, see [dbt#5095](https://github.com/dbt-labs/dbt-core/discussions/5095).
```

Reviewer 2: What does N.B. mean?

```

</VersionBlock>
<Changelog>

> - **v1.4**: Added the `--favor-state` option, which enables `--defer` to favor using the `--state` node even if the node exists in the current target. Using the `--favor-state` flag renders number 2 in following the criteria optional.

Reviewer 2: ```suggestion
- **v1.4**: Added the `--favor-state` option, which enables `--defer` to favor using the `--state` node even if the node exists in the current target. Using the `--favor-state` flag renders number 2 in the following criteria optional.
```

> - **v1.4**: Added the `--favor-state` option, which enables `--defer` to favor using the `--state` node even if the node exists in the current target. Using the `--favor-state` flag renders number 2 in the following criteria optional.

Reviewer: ```suggestion
- **v1.4**: Added the `--favor-state` option, which enables `--defer` to favor using the `--state` node even if the node exists in the current target. Using the `--favor-state` option renders the second criteria in the following optional.
```

Reviewer: - sugg for consistency for translation purposes
- sugg for maybe better readability
